src/common/WavetableScriptPresetManager.cpp

```diff
@@ -0,0 +1,275 @@
+        {
+            s->reportError("Your save path is not a directory inside the user presets directory. "
+                           "This usually means you are doing something like trying to use ../"
+                           " in your preset name.",
+                           "Invalid Save Path");
+            return;
+        }
+
+        fs::create_directories(fullLocation.parent_path());
+
+        TiXmlDeclaration decl("1.0", "UTF-8", "yes");
+
+        TiXmlDocument doc;
+        doc.InsertEndChild(decl);
+
+        TiXmlElement wts("wts");
+
+        TiXmlElement params("params");
+        params.SetAttribute("nframes", osc->wavetable_formula_nframes);
+        params.SetAttribute("res_base", osc->wavetable_formula_res_base);
+        wts.InsertEndChild(params);
+
+        TiXmlElement fm("wavetable_formula");
+        s->getPatch().wtsToXMLElement(&(s->getPatch().scene[scene].osc[oscid]), fm);
+        wts.InsertEndChild(fm);
+
+        doc.InsertEndChild(wts);
+
+        if (!doc.SaveFile(fullLocation))
+        {
+            // uhh ... do something I guess?
+            std::cout << "Could not save" << std::endl;
+        }
+
+        forcePresetRescan();
+    }
+    catch (const fs::filesystem_error &e)
+    {
+        std::ostringstream oss;
+        oss << "Exception occurred while attempting to write the preset! Most likely, "
+               "invalid characters or a reserved name was used to name the preset. "
+               "Please try again with a different name!\n"
+            << "Details " << e.what();
+        s->reportError(oss.str(), "Preset Write Error");
+    }
+}
+
+/*
+ * Given a completed path, load the preset into our storage
+ */
+void WavetableScriptPreset::loadPresetFrom(const fs::path &location, SurgeStorage *s, int scene,
+                                           int oscid)
+{
+    auto osc = &(s->getPatch().scene[scene].osc[oscid]);
+
+    TiXmlDocument doc;
+    doc.LoadFile(location);
+    auto wts = TINYXML_SAFE_TO_ELEMENT(doc.FirstChildElement("wts"));
+    if (!wts)
+    {
+        std::cout << "Unable to find wts node in document" << std::endl;
+        return;
+    }
+
+    auto params = TINYXML_SAFE_TO_ELEMENT(wts->FirstChildElement("params"));
+    if (!params)
+    {
+        std::cout << "NO PARAMS" << std::endl;
+        return;
+    }
+
+    int nframes = 0;
+    if (params->QueryIntAttribute("nframes", &nframes) == TIXML_SUCCESS)
+        osc->wavetable_formula_nframes = nframes;
+
+    int res_base = 0;
+    if (params->QueryIntAttribute("res_base", &res_base) == TIXML_SUCCESS)
+        osc->wavetable_formula_res_base = res_base;
+
+    auto frm = wts->FirstChildElement("wavetable_formula");
+    if (frm)
+        s->getPatch().wtsFromXMLElement(&(s->getPatch().scene[scene].osc[oscid]), frm);
+}
+
+/*
+ * Note: Clients rely on this being sorted by category path if you change it
+ */
+std::vector<WavetableScriptPreset::Category> WavetableScriptPreset::getPresets(SurgeStorage *s,
```

Comment:
I suppose this method is going to be the only casualty of merging .wtscripts with regular wavetables - since we already have a method in SurgeStorage to scan for WT content.